[Hub Generated] Review request for Microsoft.Security to add version stable/2020-01-01

nitsi opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/nitsi/azure-rest-api-specs/dev-security-Microsoft.Security-2020-01-01-entities-and-doc...master/).

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
Updating Microsoft.Security/stable/2020-01-01/alerts - Adding entities and updating documentation

nitsi committed:
Adds suppression to readme

nitsi committed:
Adds suppression to readme

AutorestCI commented:
/azp run automation - sdk

nitsi committed:
Adds suppression to readme

AutorestCI commented:
/azp run automation - sdk

weidongxu-microsoft commented on specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/alerts.json:
```diff
@@ -556,12 +556,596 @@
+        "type"
+      ]
+    },
+    "AccountEntity": {
```
The PR seems adding a lot of types, but I didn't see the place using them (no new API, no new property in existing type, just new types).

nitsi replied:
They are all inherit from #/definitions/AlertEntity.
They are specific implementations of that property, we need this change for 2 reasons:
1) So SDK users will have an easier time using the specific type
2) So there will be documentation of passible return values

nitsi committed:
Merge pull request #2 from Azure/master

update branch

Commit:
npm run prettier

Commit:
Fix RequiredReadOnlyProperties - Property 'type' is a required property. It should not be marked as 'readonly'.
Fix IntegerTypeMustHaveFormat
Fix R3016 - DefinitionsPropertiesNamesCamelCase
Spelling

Commit:
Fix typeo

nitsi commented:
The braking changes in this PR have no customer impact, lets review all braking changes and see why they don't have customer impact:
1) The read only property has changed from 'true' to 'false'.  
All our APIs are GET, so readonly has no meaning here. I removed it because required fields can't be readonly and fields you use for polymorphism are required.
  
2)   The new version has a different discriminator than the previous one.  
I added the discriminator, adding fields is not a braking change.

3) The new version has new required property 'type' that was not found in the old version.  
All our APIs are GET and this value was returned in all of them, it is required just for the polymorphism.

weidongxu-microsoft commented:
@nitsi 
Please contact JeffreyRichter for review on breaking changes.

Personally I think this is fine. Since `AlertEntity` previously defined with `additionalProperties`, the changes is likely safe. But I am not authorized to approval breaking changes.

JeffreyRichter commented:
We DO consider adding new properties breaking. The reason is because of SDKs. When you change the swagger, new SDKs get generated in multiple programming languages. Adding new properties may break (de)serializers in some languages and change discriminator can have a big effect on client libraries breaking customer code. 

So, we highly recommend that these changes occur in a new service api-version and then we generate new SDKs to correspond to this api-version; the new SDKs know about the new properties and discriminator.
